simplesat/transaction.py

```diff
@@ -113,23 +121,40 @@ def _safe_operations(self, pool, decisions, installed_package_ids):
                       package_id not in installed_package_ids):
                     installs.append(package_id)
 
-        # Removals should happen top down
-        for package_id in reversed(removals):
-            package = pool.id_to_package(package_id)
-            operations.append(RemoveOperation(package))
+        install_operations = []
+        remove_operations = []
 
         # Installations should happen bottom up
         for package_id in installs:
-            package = pool.id_to_package(package_id)
-            operations.append(InstallOperation(package))
+            # The solver may suggest removing a currently installed package and
+            # installing another version of that same package from a different
```

Comment:
It could mean that two teams that don't know anything about each other have published packages with the same name. The repos used within a single run don't have to come from a single universe, right? We might have two groups in a bank that have both pushed a `weekly_reports` package to their respective repositories, for example. Now a third-party user can't pull packages from both repos without running these identically named packages through the solver. So what happens then?